### PR TITLE
fixed crash caused by xml declaration inside body

### PR DESCRIPTION
**Pull request by erikdsi**
I was trying to parse a page that for some unknown reason has a xml declaration tag lost inside the body tag that caused Floki to crash.
I added a test for that edge case and a quick fix that just ignores it.

### CODE REVIEW

File: lib/floki/html_tree.ex [hunk 1]
```diff
@@ -99,14 +99,18 @@ defmodule Floki.HTMLTree do
   defp build_tree(tree, [], _, []), do: tree
   defp build_tree(tree, [{tag, attrs, child_children} | children], parent_id, stack) do
     new_id = IDSeeder.seed(tree.node_ids)
-    new_node = %HTMLNode{type: tag,
-                         attributes: attrs,
-                         node_id: new_id,
-                         parent_node_id: parent_id}
+    nodes = if tag != :pi do
```
Comment (philss) [inline lib/floki/html_tree.ex]: I think we could ignore the nodes that are `:pi` in a function match.
So, above this function declaration, we could have a declaration more or less like this:

```elixir
defp build_tree(tree, [{:pi, _, _} | children], parent_id, stack), do: build_tree(tree, children, parent_id, stack)
```

This version would avoid an `if`. What do you think?

Comment (sabondano) [inline lib/floki/html_tree.ex]: Hey, I was running into similar issues. Wondering if you know what "pi" stands for? Just curious.

Comment (erikdsi) [inline lib/floki/html_tree.ex]: I implemented, tested, works and is definitely a more idiomatic way to solve it.

File: lib/floki/html_tree.ex [hunk 2]
```diff
@@ -97,16 +97,18 @@ defmodule Floki.HTMLTree do
   defp get_ids_for_delete_stack(_), do: []
 
   defp build_tree(tree, [], _, []), do: tree
+  defp build_tree(tree, [{:pi, _, _} | children], parent_id, stack), do: build_tree(tree, children, parent_id, stack)
```
Comment (sabondano) [inline lib/floki/html_tree.ex]: @erikdsi do you know what "pi" stands for?

Comment (mischov) [inline lib/floki/html_tree.ex]: processing instruction

Comment (sabondano) [inline lib/floki/html_tree.ex]: Thank you :)

**Comment by philss** [comment]
@erikdsi Thank you! 😃

**Comment by philss** [comment]
I forgot to mention earlier: this fix is available in version `0.17.1` and above.
